### PR TITLE
Support list es nodes

### LINKED ISSUE #5635: Elasticsearch exporter: List urls for connection
I want to use list urls for configure es exporter, because i have es with some nodes

As in es`s docs - https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high-getting-started-initialization.html

### PR DESCRIPTION
**Pull request by jooble**
## Description

Allows use list urls in exporters` configuration

## Related issues

closes #5635 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [x] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 9063229fcf934f4f50995915f31e38bb1ae93778
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by liljaylj** [comment]
LGTM. Is it going to be merged?

File: exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java [hunk 1]
```diff
@@ -204,4 +204,26 @@ public void shouldFlushOnMemoryLimit() {
     // then
     assertThat(client.shouldFlush()).isTrue();
   }
+
+  @Test
+  public void shouldUseLegacyUrlIfExist() {
```
Comment (npepinpe) [inline exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java:209]: As you can see from the other tests in the file, we tend to use a format of

```java
@Test
public void shouldTestSomething() {
  // given
  // add all the setup here, i.e. variable declaration, object instantiation, etc.
  
  // when
  // add the behaviour you want to test: for example, here, creating the client is the behaviour under test.
  
  // then
  // add your assertions here - in this case, something like the following:
  assertThat(client.client.getNodes().stream().map(Node::getHost)).containsExactly(elastic.getRestHttpHosts()[0]);
}
```

I would also argue a test with no assertions (just checking that an exception is thrown) is typically a code smell - what if we stop resolving the host at initialization and start doing so lazily? Then the test breaks. We should aim for explicit asserts as much as possible.

File: exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java [hunk 2]
```diff
@@ -204,4 +204,26 @@ public void shouldFlushOnMemoryLimit() {
...
+    configuration = new ElasticsearchExporterConfiguration();
+    configuration.url = "http://localhost:8080";
```
Comment (npepinpe) [inline exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java:211]: We can use `elastic.getHttpHosts()[0].toString()`

File: exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java [hunk 3]
```diff
@@ -204,4 +204,26 @@ public void shouldFlushOnMemoryLimit() {
...
+    logSpy = spy(LoggerFactory.getLogger(ElasticsearchClientTest.class));
+
+    client = new ElasticsearchClient(configuration, logSpy);
+  }
+
+  @Test(expected = ElasticsearchExporterException.class)
+  public void shouldUseUrlslIfUrlNotExist() {
+    configuration = new ElasticsearchExporterConfiguration();
+    configuration.url = null;
+    configuration.urls = List.of("bad url");
```
Comment (npepinpe) [inline exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java:223]: There's no need to expect an exception here - we can simply create a list of fake URLs, e.g.

```java
configuration.urls = List.of("http://localhost:8080", "http://localhost:8081");
```

And assert the following:

```java
assertThat(client.client.getNodes().stream().map(Node::getHost).map(HttpHost::toString))
        .containsExactlyInAnyOrderElementsOf(configuration.urls);
```

**Review by npepinpe — CHANGES_REQUESTED** [review]
Thanks for your contribution, and sorry for the slow review - this fell through the cracks, then the holidays were on us. I understand if you've moved on, let us know if we should take it over and finish it for you :+1: 

Regarding the tests added in `ElasticsearchClientTest` - this class is misnamed, as these are integration tests, whereas the tests added are simple unit tests. Please rename it `ElasticsearchClientIT`, and create a new `ElasticsearchClientTest` which then includes only your two tests (and in the future, tests which just check the client is well constructed, or at least do not rely on a real Elasticsearch node).

**Comment by liljaylj** [comment]
@jooble , found a problem - it is not possible to set `urls` using environment variable `ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_URLS`.

my proposal is to set urls using comma-separated host list, then split, trim and map and convert to `HttpHost[]`

**Comment by npepinpe** [comment]
Superseded by https://github.com/zeebe-io/zeebe/pull/6467

**Pull request closed without merging**